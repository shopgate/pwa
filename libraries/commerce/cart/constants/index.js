export const CART_PATH = '/cart';
export const DEEPLINK_CART_ADD_PRODUCT_PATTERN = '/cart_add_product/:productId/:couponId';
export const DEEPLINK_CART_ADD_COUPON_PATTERN = '/cart_add_coupon/:couponId';

export const CART_ITEM_TYPE_COUPON = 'coupon';
export const CART_ITEM_TYPE_PRODUCT = 'product';
export const CART_TOTALS_TYPE_SUB = 'subTotal';
export const CART_TOTALS_TYPE_GRAND = 'grandTotal';
export const CART_TOTALS_TYPE_DISCOUNT = 'discount';
export const CART_TOTALS_TYPE_SHIPPING = 'shipping';
export const CART_TOTALS_TYPE_TAX = 'tax';
export const COUPON_TYPE_FIXED = 'fixed';
export const COUPON_TYPE_PERCENTAGE = 'percentage';

export const MESSAGE_TYPE_ERROR = 'error';

export const ADD_PRODUCTS_TO_CART = 'ADD_PRODUCTS_TO_CART';
export const SUCCESS_ADD_PRODUCTS_TO_CART = 'SUCCESS_ADD_PRODUCTS_TO_CART';
export const ERROR_ADD_PRODUCTS_TO_CART = 'ERROR_ADD_PRODUCTS_TO_CART';
export const DELETE_PRODUCTS_FROM_CART = 'DELETE_PRODUCTS_FROM_CART';
export const ERROR_DELETE_PRODUCTS_FROM_CART = 'ERROR_DELETE_PRODUCTS_FROM_CART';
export const SUCCESS_DELETE_PRODUCTS_FROM_CART = 'SUCCESS_DELETE_PRODUCTS_FROM_CART';
export const UPDATE_PRODUCTS_IN_CART = 'UPDATE_PRODUCTS_IN_CART';
export const SUCCESS_UPDATE_PRODUCTS_IN_CART = 'SUCCESS_UPDATE_PRODUCTS_IN_CART';
export const ERROR_UPDATE_PRODUCTS_IN_CART = 'ERROR_UPDATE_PRODUCTS_IN_CART';
export const ADD_COUPONS_TO_CART = 'ADD_COUPONS_TO_CART';
export const SUCCESS_ADD_COUPONS_TO_CART = 'SUCCESS_ADD_COUPONS_TO_CART';
export const ERROR_ADD_COUPONS_TO_CART = 'ERROR_ADD_COUPONS_TO_CART';
export const DELETE_COUPONS_FROM_CART = 'DELETE_COUPONS_FROM_CART';
export const ERROR_DELETE_COUPONS_FROM_CART = 'ERROR_DELETE_COUPONS_FROM_CART';
export const SET_CART_PENDING_PRODUCT_COUNT = 'SET_CART_PENDING_PRODUCT_COUNT';
export const SUCCESS_DELETE_COUPONS_FROM_CART = 'SUCCESS_DELETE_COUPONS_FROM_CART';

export const REQUEST_CART = 'REQUEST_CART';
export const RECEIVE_CART = 'RECEIVE_CART';
export const ERROR_CART = 'ERROR_CART';

/**
 * Event that is called when addToCart failed because not all variants where selected yet.
 * @type {string}
 */
export const EVENT_ADD_TO_CART_MISSING_VARIANT = 'EVENT_ADD_TO_CART_MISSING_VARIANT';
