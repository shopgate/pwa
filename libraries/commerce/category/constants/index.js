export const CATEGORY_PATH = '/category';
export const ROOT_CATEGORY_PATTERN = CATEGORY_PATH;
export const CATEGORY_PATTERN = `${CATEGORY_PATH}/:categoryId`;
export const CATEGORY_ALL_PATTERN = `${CATEGORY_PATH}/:categoryId/all`;
export const CATEGORY_FILTER_PATTERN = `${CATEGORY_PATH}/:categoryId/filter`;

export const CATEGORY_LIFETIME = 3600000; // 1 hour in milliseconds

export const REQUEST_ROOT_CATEGORIES = 'REQUEST_ROOT_CATEGORIES';
export const RECEIVE_ROOT_CATEGORIES = 'RECEIVE_ROOT_CATEGORIES';
export const ERROR_ROOT_CATEGORIES = 'ERROR_ROOT_CATEGORIES';

export const REQUEST_CATEGORY = 'REQUEST_CATEGORY';
export const RECEIVE_CATEGORY = 'RECEIVE_CATEGORY';
export const ERROR_CATEGORY = 'ERROR_CATEGORY';

export const REQUEST_CATEGORY_CHILDREN = 'REQUEST_CATEGORY_CHILDREN';
export const RECEIVE_CATEGORY_CHILDREN = 'RECEIVE_CATEGORY_CHILDREN';
export const ERROR_CATEGORY_CHILDREN = 'ERROR_CATEGORY_CHILDREN';
