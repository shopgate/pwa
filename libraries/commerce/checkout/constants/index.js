export const LEGACY_URL = '/checkout_legacy';
export const FETCH_CHECKOUT_URL_TIMEOUT = 12000; // 12 seconds timeout for fetching a checkout url.
export const SUCCESS_CHECKOUT = 'SUCCESS_CHECKOUT';
