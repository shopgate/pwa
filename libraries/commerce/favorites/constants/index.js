export const FAVORITES_PATH = '/favourite_list';

export const ADD_PRODUCT_TO_FAVORITES = 'ADD_PRODUCT_TO_FAVORITES';
export const REMOVE_PRODUCT_FROM_FAVORITES = 'REMOVE_PRODUCT_FROM_FAVORITES';

export const REQUEST_FAVORITES = 'REQUEST_FAVORITES';
export const RECEIVE_FAVORITES = 'RECEIVE_FAVORITES';
export const ERROR_FAVORITES = 'ERROR_FAVORITES';
export const ERROR_FETCH_FAVORITES = 'ERROR_FETCH_FAVORITES';

export const REQUEST_ADD_FAVORITES = 'REQUEST_ADD_FAVORITES';
export const SUCCESS_ADD_FAVORITES = 'SUCCESS_ADD_FAVORITES';
export const ERROR_ADD_FAVORITES = 'ERROR_ADD_FAVORITES';

export const REQUEST_REMOVE_FAVORITES = 'REQUEST_REMOVE_FAVORITES';
export const SUCCESS_REMOVE_FAVORITES = 'SUCCESS_REMOVE_FAVORITES';
export const ERROR_REMOVE_FAVORITES = 'ERROR_REMOVE_FAVORITES';

export const CANCEL_REQUEST_SYNC_FAVORITES = 'CANCEL_REQUEST_SYNC_FAVORITES';

export const IDLE_SYNC_FAVORITES = 'IDLE_SYNC_FAVORITES';
export const REQUEST_FLUSH_FAVORITES_BUFFER = 'REQUEST_FLUSH_FAVORITES_BUFFER';

// Defines a local error code which is not related to a backend call
export const FAVORITES_LIMIT_ERROR = 'FAVORITES_LIMIT_ERROR';

export const FAVORITES_LIFETIME = 3600000; // 1 hour
export const FETCH_FAVORITES_THROTTLE = (process.env && process.env.NODE_ENV === 'test') ? 0 : 2000;
export const FAVORITE_ACTION_BUFFER_TIME = 500;
export const FAVORITE_BUTTON_DEBOUNCE_TIME = 200; // Handles duplicated add/remove button clicks

export const RECEIVE_FAVORITES_LISTS = 'SUCCESS_FETCH_FAVORITES_LIST';
export const SUCCESS_ADD_FAVORITES_LIST = 'SUCCESS_ADD_FAVORITES_LIST';
export const SUCCESS_UPDATE_FAVORITES_LIST = 'SUCCESS_UPDATE_FAVORITES_LIST';
export const SUCCESS_REMOVE_FAVORITES_LIST = 'SUCCESS_REMOVE_FAVORITES_LIST';

export const OPEN_FAVORITE_LIST_CHOOSER = 'OPEN_FAVORITE_LIST_CHOOSER';
export const CLOSE_FAVORITE_LIST_CHOOSER = 'CLOSE_FAVORITE_LIST_CHOOSER';
