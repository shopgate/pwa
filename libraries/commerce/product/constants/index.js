export const ITEM_PATH = '/item';
export const ITEM_PATTERN = `${ITEM_PATH}/:productId`;
export const ITEM_GALLERY_PATTERN = `${ITEM_PATH}/:productId/gallery/:slide`;
export const ITEM_REVIEWS_PATTERN = `${ITEM_PATH}/:productId/reviews`;
export const ITEM_WRITE_REVIEW_PATTERN = `${ITEM_PATH}/:productId/write_review`;

export const PRODUCT_LIFETIME = 3600000; // 1 hour in milliseconds

// PRODUCT
export const REQUEST_PRODUCT = 'REQUEST_PRODUCT';
export const RECEIVE_PRODUCT = 'RECEIVE_PRODUCT';
export const ERROR_PRODUCT = 'ERROR_PRODUCT';
// PRODUCTS
export const REQUEST_PRODUCTS = 'REQUEST_PRODUCTS';
export const RECEIVE_PRODUCTS = 'RECEIVE_PRODUCTS';
export const RECEIVE_PRODUCT_CACHED = 'RECEIVE_PRODUCTS_CACHED';
export const ERROR_PRODUCTS = 'ERROR_PRODUCTS';
// PRODUCT VARIANTS
export const REQUEST_PRODUCT_VARIANTS = 'REQUEST_PRODUCT_VARIANTS';
export const RECEIVE_PRODUCT_VARIANTS = 'RECEIVE_PRODUCT_VARIANTS';
export const ERROR_PRODUCT_VARIANTS = 'ERROR_PRODUCT_VARIANTS';
// PRODUCT OPTIONS
export const REQUEST_PRODUCT_OPTIONS = 'REQUEST_PRODUCT_OPTIONS';
export const RECEIVE_PRODUCT_OPTIONS = 'RECEIVE_PRODUCT_OPTIONS';
export const ERROR_PRODUCT_OPTIONS = 'ERROR_PRODUCT_OPTIONS';
// PRODUCT DESCRIPTION
export const REQUEST_PRODUCT_DESCRIPTION = 'REQUEST_PRODUCT_DESCRIPTION';
export const RECEIVE_PRODUCT_DESCRIPTION = 'RECEIVE_PRODUCT_DESCRIPTION';
export const ERROR_PRODUCT_DESCRIPTION = 'ERROR_PRODUCT_DESCRIPTION';
// PRODUCT PROPERTIES
export const REQUEST_PRODUCT_PROPERTIES = 'REQUEST_PRODUCT_PROPERTIES';
export const RECEIVE_PRODUCT_PROPERTIES = 'RECEIVE_PRODUCT_PROPERTIES';
export const ERROR_PRODUCT_PROPERTIES = 'ERROR_PRODUCT_PROPERTIES';
// PRODUCT SHIPPING
export const REQUEST_PRODUCT_SHIPPING = 'REQUEST_PRODUCT_SHIPPING';
export const RECEIVE_PRODUCT_SHIPPING = 'RECEIVE_PRODUCT_SHIPPING';
export const ERROR_PRODUCT_SHIPPING = 'ERROR_PRODUCT_SHIPPING';
// PRODUCT IMAGES
export const REQUEST_PRODUCT_IMAGES = 'REQUEST_PRODUCT_IMAGES';
export const RECEIVE_PRODUCT_IMAGES = 'RECEIVE_PRODUCT_IMAGES';
export const ERROR_PRODUCT_IMAGES = 'ERROR_PRODUCT_IMAGES';
// CURRENT PRODUCT
export const RESET_CURRENT_PRODUCT = 'RESET_CURRENT_PRODUCT';

export const EXPIRE_PRODUCT_BY_ID = 'EXPIRE_PRODUCT_BY_ID';
export const SET_PRODUCT_ID = 'SET_PRODUCT_ID';
export const SET_PRODUCT_VARIANT_ID = 'SET_PRODUCT_VARIANT_ID';
export const SET_PRODUCT_QUANTITY = 'SET_PRODUCT_QUANTITY';
export const SET_PRODUCT_OPTION = 'SET_PRODUCT_OPTION';

// PRODUCT AVAILABILITY STATES
export const AVAILABILITY_STATE_OK = 'ok';
export const AVAILABILITY_STATE_WARNING = 'warning';
export const AVAILABILITY_STATE_ALERT = 'alert';

export const OPTION_TYPE_SELECT = 'select';
export const OPTION_TYPE_TEXT = 'text';

export const PROPERTIES_FILTER_WHITELIST = 'whitelist';
export const PROPERTIES_FILTER_BLACKLIST = 'blacklist';

// PRODUCT RELATIONS
export const REQUEST_PRODUCT_RELATIONS = 'REQUEST_PRODUCT_RELATIONS';
export const RECEIVE_PRODUCT_RELATIONS = 'RECEIVE_PRODUCT_RELATIONS';
export const ERROR_PRODUCT_RELATIONS = 'ERROR_PRODUCT_RELATIONS';
export const PRODUCT_RELATIONS_DEFAULT_LIMIT = 20;

// PRODUCT RELATIONS TYPES
export const PRODUCT_RELATIONS_TYPE_CROSS_SELLING = 'crossSelling';
export const PRODUCT_RELATIONS_TYPE_UPSELLING = 'upselling';
export const PRODUCT_RELATIONS_TYPE_BONUS = 'bonus';
export const PRODUCT_RELATIONS_TYPE_BOUGHT_WITH = 'boughtWith';
export const PRODUCT_RELATIONS_TYPE_CUSTOM = 'custom';

// PRODUCT METADATA
export const UPDATE_METADATA = 'UPDATE_METADATA';
