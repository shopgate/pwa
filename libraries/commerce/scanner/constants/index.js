export const SCANNER_STARTED = 'SCANNER_STARTED';
export const SCANNER_CANCELLED = 'SCANNER_CANCELLED';
export const SCANNER_FINISHED = 'SCANNER_FINISHED';

export const SUCCESS_HANDLE_SCANNER = 'SUCCESS_HANDLE_SCANNER';
export const ERROR_HANDLE_SCANNER = 'ERROR_HANDLE_SCANNER';

export const SCANNER_PATH = '/scanner';

export const QR_CODE_TYPE_COUPON = '2';
export const QR_CODE_TYPE_HOMEPAGE = '5';
export const QR_CODE_TYPE_PRODUCT = '7';
export const QR_CODE_TYPE_PRODUCT_WITH_COUPON = '9';
export const QR_CODE_TYPE_CATEGORY = 'e';
export const QR_CODE_TYPE_SEARCH = 'b';
export const QR_CODE_TYPE_PAGE = '1';

export const QR_CODE_HOST_2DS = '2d.is';

export const SCANNER_FORMATS_BARCODE = [
  'UPC_E',
  'CODE_39',
  'EAN_13',
  'EAN_8',
  'CODE_93',
  'CODE_128',
  'PDF_417',
  'ITF',
];

export const SCANNER_FORMATS_QR_CODE = ['QR_CODE'];
