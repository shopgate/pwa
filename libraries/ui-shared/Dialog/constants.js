export const DIALOG_TEXT_MESSAGE = 'DIALOG_TEXT_MESSAGE';
export const DIALOG_HTML_CONTENT = 'DIALOG_HTML_CONTENT';
export const MODAL_VARIANT_SELECT = 'MODAL_VARIANT_SELECT';

export const MODAL_ACTION_TYPE_NORMAL = 'normal';
export const MODAL_ACTION_TYPE_PRIMARY = 'primary';
export const MODAL_ACTION_TYPE_DANGER = 'danger';
