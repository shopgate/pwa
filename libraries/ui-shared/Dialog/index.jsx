import React from 'react';
import PropTypes from 'prop-types';
import Modal from '@shopgate/pwa-common/components/Modal';
import Backdrop from '@shopgate/pwa-common/components/Backdrop';
import { MODAL_PIPELINE_ERROR } from '@shopgate/pwa-common/constants/ModalTypes';
import I18n from '@shopgate/pwa-common/components/I18n';
import {
  DIALOG_TEXT_MESSAGE,
  MODAL_VARIANT_SELECT,
  MODAL_ACTION_TYPE_PRIMARY,
  MODAL_ACTION_TYPE_NORMAL,
  DIALOG_HTML_CONTENT,
} from './constants';
import PipelineErrorDialog from './components/PipelineErrorDialog';
import TextMessageDialog from './components/TextMessageDialog';
import HtmlContentDialog from './components/HtmlContentDialog';
import BasicDialog from './components/BasicDialog';
import VariantSelectModal from './components/VariantSelectModal';

const dialogTypes = {
  [DIALOG_TEXT_MESSAGE]: TextMessageDialog,
  [DIALOG_HTML_CONTENT]: HtmlContentDialog,
  [MODAL_PIPELINE_ERROR]: PipelineErrorDialog,
  [MODAL_VARIANT_SELECT]: VariantSelectModal,
};

/**
 * The main component for rendering dialogs.
 * This component takes care of choosing the correct component body for the given type
 * and render it on a modal overlay.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
const Dialog = ({
  modal, onConfirm, onDismiss, children,
}) => {
  // Assemble the actions.
  const actions = [];
  const {
    confirm,
    dismiss,
    title,
    titleParams,
    message,
    params,
    type,
    content,
    confirmDisabled,
    dismissDisabled,
  } = modal;

  // Push dismiss action first so the button is rendered first
  if (dismiss) {
    // We have a dismiss label, add a dismiss action.
    actions.push({
      label: modal.dismiss,
      action: onDismiss,
      type: MODAL_ACTION_TYPE_PRIMARY,
      disabled: dismissDisabled,
    });
  }

  if (confirm) {
    // We have a confirm label, add a confirm action.
    actions.push({
      label: modal.confirm,
      action: onConfirm,
      type: MODAL_ACTION_TYPE_NORMAL,
      disabled: confirmDisabled,
    });
  }

  let dialogType = type;

  if (!dialogType && message) {
    dialogType = DIALOG_TEXT_MESSAGE;
  }

  let dialogTitle = title;
  if (titleParams) {
    dialogTitle = <I18n.Text string={title} params={titleParams} />;
  }

  const dialogProps = {
    actions,
    title: dialogTitle,
    params,
    message: message || undefined,
    children: content || children,
  };

  const DialogComponent = dialogTypes[dialogType] || BasicDialog;

  return (
    <Modal>
      <Backdrop isVisible level={0} />
      <DialogComponent {...dialogProps} />
    </Modal>
  );
};

Dialog.propTypes = {
  modal: PropTypes.shape({
    title: BasicDialog.propTypes.title,
    titleParams: PropTypes.shape(),
    confirm: PropTypes.string,
    dismiss: PropTypes.string,
    message: PropTypes.string,
    params: PropTypes.shape(),
    type: PropTypes.string,
    content: PropTypes.node,
    /**
     * Whether the confirm button is disabled when visible
     */
    confirmDisabled: PropTypes.bool,
    /**
     * Whether the dismiss button is disabled when visible
     */
    dismissDisabled: PropTypes.bool,
  }).isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  children: PropTypes.any,
  onConfirm: PropTypes.func,
  onDismiss: PropTypes.func,
};

Dialog.defaultProps = {
  onConfirm: () => {},
  onDismiss: () => {},
  children: null,
};

export default Dialog;
