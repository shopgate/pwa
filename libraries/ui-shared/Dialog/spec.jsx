import React from 'react';
import { shallow, ReactWrapper } from 'enzyme';
import { MODAL_PIPELINE_ERROR } from '@shopgate/pwa-common/constants/ModalTypes';
import { MODAL_VARIANT_SELECT } from './constants';
import Dialog from './index';

jest.mock('./components/VariantSelectModal', () => {
  /**
   * VariantSelectModal mock.
   * @return {JSX}
   */
  const VariantSelectModal = () => <div />;
  return VariantSelectModal;
});

jest.mock('@shopgate/engage/components');

describe('<Dialog />', () => {
  it('should render without props', () => {
    const wrapper = shallow(<Dialog modal={{ message: 'msg' }} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('TextMessageDialog').length).toBe(1);
  });

  it('should render BasicDialog when no message given', () => {
    const wrapper = shallow(<Dialog modal={{ message: null }} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('BasicDialog').length).toBe(1);
  });

  it('should render a special dialog', () => {
    const params = {
      errorCode: '',
      message: '',
      pipeline: '',
      request: {},
    };

    const wrapper = shallow(<Dialog
      modal={{
        type: MODAL_PIPELINE_ERROR,
        params,
      }}
    />);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('DefaultDialog').length).toBe(0);
    expect(wrapper.find('PipelineErrorDialog').length).toBe(1);
  });

  it('should render variant select dialog', () => {
    const params = {
      productId: 'product_1',
    };

    const wrapper = shallow(<Dialog
      modal={{
        message: 'Test',
        type: MODAL_VARIANT_SELECT,
        params,
      }}
    />);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('DefaultDialog').length).toBe(0);
    expect(wrapper.find('VariantSelectModal').length).toBe(1);
  });

  it('should convert title into translatable element', () => {
    const title = 'translate.me';
    const titleParams = {
      foo: 'bar',
    };

    // eslint-disable-next-line extra-rules/no-single-line-objects
    const wrapper = shallow(<Dialog modal={{ title, titleParams }} />);
    const i18n = new ReactWrapper(wrapper.find('BasicDialog').prop('title'));

    expect(i18n.prop('string')).toEqual(title);
    expect(i18n.prop('params')).toEqual(titleParams);
  });
});
