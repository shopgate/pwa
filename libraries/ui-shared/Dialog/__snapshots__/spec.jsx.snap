// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> should render BasicDialog when no message given 1`] = `
<Modal>
  <Backdrop
    className=""
    color="#000"
    duration={200}
    isVisible={true}
    level={0}
    onClick={[Function]}
    opacity={50}
  />
  <BaseDialog
    actions={Array []}
    title={null}
  />
</Modal>
`;

exports[`<Dialog /> should render a special dialog 1`] = `
<Modal>
  <Backdrop
    className=""
    color="#000"
    duration={200}
    isVisible={true}
    level={0}
    onClick={[Function]}
    opacity={50}
  />
  <PipelineErrorDialog
    actions={Array []}
    message=""
    params={
      Object {
        "errorCode": "",
        "message": "",
        "pipeline": "",
        "request": Object {},
      }
    }
  />
</Modal>
`;

exports[`<Dialog /> should render variant select dialog 1`] = `
<Modal>
  <Backdrop
    className=""
    color="#000"
    duration={200}
    isVisible={true}
    level={0}
    onClick={[Function]}
    opacity={50}
  />
  <VariantSelectModal
    actions={Array []}
    message="Test"
    params={
      Object {
        "productId": "product_1",
      }
    }
  />
</Modal>
`;

exports[`<Dialog /> should render without props 1`] = `
<Modal>
  <Backdrop
    className=""
    color="#000"
    duration={200}
    isVisible={true}
    level={0}
    onClick={[Function]}
    opacity={50}
  />
  <TextMessageDialog
    actions={Array []}
    message="msg"
    params={Object {}}
    title={null}
  />
</Modal>
`;
